content/blogs/2024-03-04-kestra-databricks.md

```diff
@@ -0,0 +1,159 @@
+
+### Running Query on Databricks ###
+
+Let us now perform a couple of query operations on Databricks. Kestra’s Databricks plugin provides the [query task](https://kestra.io/plugins/tasks/sql/io.kestra.plugin.databricks.sql.query) for querying data from Databricks. This task requires you to mention the `httpPath`. In order to get the `httpPath`, you need to go to the compute cluster using which you need to perform this query. On the compute cluster’s page, go to `JDBC/ODBC` tab under the Advanced options. This is where you will get the `HTTP Path` which you can use in the `httpPath` attribute of your query task.
+
+The first query task that we will perform will be to load the data from the DBFS file onto a table. The query in this task will read the CSV file from the DBFS. It will create the table in the Databricks by inferring the schema from the CSV file, and the data will be filled in the table from this CSV file.
+
+The following code snippet will create the table, and load the data from the CSV file present in the DBFS.
+
+```yaml
+id: databricks-load-table
+namespace: dev
+tasks:
+  - id: load-table-from-dbfs
+    type: io.kestra.plugin.databricks.sql.Query
+    accessToken: <your-token>
+    host: <your-host>
+    httpPath: <your-httpPath>
+    sql: "CREATE TABLE orders USING CSV LOCATION '/Shared/kestra/data/orders.csv' OPTIONS (header \"true\", inferSchema \"true\");"
+```
+
+Once you run the above flow, you can go to the Catalog section on the Databricks console, and check the newly created table in the default catalog and schema of your Databricks workspace.
+
+Let us now move on to the second query task where we will be querying this newly created table. We will query the orders table to get the top 10 orders in terms of the highest order total.
+
+The following code snippet will query the orders table to get the top 10 orders.
+
+```yaml
+id: databricks-query
+namespace: dev
+tasks:
+  - id: query-top-10-orders
+    type: io.kestra.plugin.databricks.sql.Query
+    accessToken: <your-token>
+    host: <your-host>
+    httpPath: <your-httpPath>
+    sql: "SELECT * FROM orders ORDER BY total DESC limit 10;"
+```
+
+On running this flow, take a look at the Outputs tab of this execution. The Outputs tab would contain the `size` indicating the number of records fetched by the query, and an `uri` to the Kestra’s internal storage file which contains the fetched records from the query.
+
+![top_10_orders_query_output](/blogs/2024-03-04-kestra-databricks/top_10_orders_query_output.png)
+
+### Delete Compute Cluster ###
+
+Now that we have performed all the tasks on the Databricks cluster, let us delete the compute cluster. Kestra’s Databricks plugin provides the [DeleteCluster task](https://kestra.io/plugins/tasks/cluster/io.kestra.plugin.databricks.cluster.deletecluster)  that will delete the compute cluster. This task takes the compute cluster’s clusterId in order to pick the corresponding cluster for deletion.
```

Comment:
```suggestion
Now that we have performed all the tasks on the Databricks cluster, we can delete it to avoid unnecessary compute costs. Kestra’s Databricks plugin provides the [DeleteCluster task](https://kestra.io/plugins/tasks/cluster/io.kestra.plugin.databricks.cluster.deletecluster)  that will delete the compute cluster. This task takes the `clusterId` to pick the corresponding cluster for deletion.
```